test/test_quantized_convolutional_layer.h

```diff
@@ -97,97 +93,141 @@ TEST(quantized_convolutional, fprop) {
-    tensor_t in_tensor = create_simple_tensor(25)
-        , out_tensor = create_simple_tensor(18)
-        , a_tensor = create_simple_tensor(18)
-        , weight_tensor = create_simple_tensor(18)
-        , bias_tensor = create_simple_tensor(2);
-
-    // short-hand references to the payload vectors
-    vec_t &in = in_tensor[0]
-        , &out = out_tensor[0]
-        , &weight = weight_tensor[0];
-
-    ASSERT_EQ(l.in_shape()[1].size(), serial_size_t(18)); // weight
-
-    uniform_rand(in.begin(), in.end(), -1.0, 1.0);
-
-    std::vector<tensor_t*> in_data, out_data;
-    in_data.push_back(&in_tensor);
-    in_data.push_back(&weight_tensor);
-    in_data.push_back(&bias_tensor);
-    out_data.push_back(&out_tensor);
-    out_data.push_back(&a_tensor);
-    l.setup(false);
-    {
-        l.forward_propagation(in_data, out_data);
+  // create simple tensors that wrap the payload vectors of the correct size
+  tensor_t in_tensor     = create_simple_tensor(25),
+           out_tensor    = create_simple_tensor(18),
+           weight_tensor = create_simple_tensor(18),
+           bias_tensor   = create_simple_tensor(2);
 
-        for (auto o: out)
-            EXPECT_NEAR(0.5, o, 1E-3);
-    }
+  // short-hand references to the payload vectors
+  vec_t &in = in_tensor[0], &out = out_tensor[0], &weight = weight_tensor[0];
 
-    weight[0] = 0.3;  weight[1] = 0.1; weight[2] = 0.2;
-    weight[3] = 0.0;  weight[4] =-0.1; weight[5] =-0.1;
-    weight[6] = 0.05; weight[7] =-0.2; weight[8] = 0.05;
+  ASSERT_EQ(l.in_shape()[1].size(), serial_size_t(18));  // weight
 
-    weight[9]  = 0.0; weight[10] =-0.1; weight[11] = 0.1;
-    weight[12] = 0.1; weight[13] =-0.2; weight[14] = 0.3;
-    weight[15] = 0.2; weight[16] =-0.3; weight[17] = 0.2;
+  uniform_rand(in.begin(), in.end(), -1.0, 1.0);
 
-    in[0] = 3;  in[1] = 2;  in[2] = 1;  in[3] = 5; in[4] = 2;
-    in[5] = 3;  in[6] = 0;  in[7] = 2;  in[8] = 0; in[9] = 1;
-    in[10] = 0; in[11] = 6; in[12] = 1; in[13] = 1; in[14] = 10;
-    in[15] = 3; in[16] =-1; in[17] = 2; in[18] = 9; in[19] = 0;
-    in[20] = 1; in[21] = 2; in[22] = 1; in[23] = 5; in[24] = 5;
+  std::vector<tensor_t *> in_data, out_data;
+  in_data.push_back(&in_tensor);
+  in_data.push_back(&weight_tensor);
+  in_data.push_back(&bias_tensor);
+  out_data.push_back(&out_tensor);
+  l.setup(false);
+  {
+    l.forward_propagation(in_data, out_data);
 
-    {
-        l.forward_propagation(in_data, out_data);
+    for (auto o : out) EXPECT_NEAR(0.5, o, 1E-3);
+  }
 
-        EXPECT_NEAR(0.4875026, out[0], 2E-2);
-        EXPECT_NEAR(0.8388910, out[1], 2E-2);
-        EXPECT_NEAR(0.8099984, out[2], 2E-2);
-        EXPECT_NEAR(0.7407749, out[3], 2E-2);
-        EXPECT_NEAR(0.5000000, out[4], 2E-2);
-        EXPECT_NEAR(0.1192029, out[5], 2E-2);
-        EXPECT_NEAR(0.5986877, out[6], 2E-2);
-        EXPECT_NEAR(0.7595109, out[7], 2E-2);
-        EXPECT_NEAR(0.6899745, out[8], 2E-2);
-    }
+  weight[0] = 0.3;
```

Comment:
Please, change to compact formattion, and add guard comments ` // clang-format off `